src/idl_gen_python.cpp

```diff
@@ -991,6 +999,9 @@ class PythonGenerator : public BaseGenerator {
         default:
           // Scalar or sting fields.
           field_type = GetBasePythonTypeForScalarAndString(base_type);
+          if (field.IsScalarOptional()) {
```

Comment:
I think this logic belongs in`GetBasePythonTypeForScalarAndString`, based on the name, wdyt?

also what a long method name lol

Reply:
I originally wanted to do that, but that method is also used for generating the type name of arrays, which don't have an associated schema field to pass to the function.  Similarly, array elements can't be optional.